app/models/slack_bot.rb

```diff
@@ -3,7 +3,7 @@ class SlackBot < ActiveRecord::Base
 
   def self.from_auth_hash(auth_hash)
     bot_info = auth_hash[:extra][:bot_info]
-    bot = find_by(id: bot_info[:bot_user_id])
+    bot = find_by(id: bot_info[:bot_access_token])
```

Comment:
Lol, I see what you mean, in the else part of the statement, if we don't find the bot per id, we create a new record with the id being the access token... weird.

Reply:
We always expect bot_info to contains the access_token, right? We don't ever want to create a bot with a `nil` id? In this case we should be explicit and failing hard if we don't get a token
```suggestion
    bot = find_by(id: bot_info.fetch(:bot_access_token))
```

Reply:
Reason being, in ruby, `bot_info[:bot_access_token]` will return nil, if bot_info doesn't have a `bot_access_token` key in its hash.

With `bot_info.fetch(:bot_access_token)`, in the key isn't present, ruby will raise a KeyError exception, and we'll know immediately that something is missing.

It's bad practice in ruby not to use `fetch` for accessing any hash keys, and if you're ok with them not being present, you can provide a default as such: `bot_info.fetch(:bot_access_token, nil)`.

I know this codebase is riddled with accessing keys in hashes that may or may not be present, and I won't make you update all of them right now, but just something good to know :)

Reply:
> We always expect bot_info to contains the access_token, right? We don't ever want to create a bot with a nil id? In this case we should be explicit and failing hard if we don't get a token

I believe so. `id` is the primary key, so it'll fail at insert, but I agree with you.

Great, this is exactly the sort of review I was hoping for! Thanks!